aten/src/ATen/native/CPUFallback.cpp

```diff
@@ -135,18 +141,46 @@ void cpu_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
           TORCH_CHECK(found_alias, "The operator ", op.schema().operator_name(), " appears to have invalid alias information. ",
                       "Found a return tensor argument with a mismatched mutable alias: ", schema_returns[idx]);
         } else {
+          // Decide what device to move the output tensor(s) to.
+          // The current convention is that we use the first tensor arg to pick the device
+          // Barring that, we take the first tensor from a TensorList arg.
+          c10::optional<c10::Device> tgt_device = c10::nullopt;
+          if (tensor_args.size() > 0) {
+              tgt_device = tensor_args[0].device();
+          } else {
+            // We need to loop through all of the (potentially multiple) TensorList arguments
+            // In case, e.g. the first one is empty but the second is not.
+            for (auto& tens_list : tensorlist_args) {
```

Comment:
This should go in a helper, it's too hard to tell right away what these nested loops are doing.